src/accountsdb/buffer_pool.zig

```diff
@@ -296,247 +241,299 @@ pub const BufferPool = struct {
         file_offset_start: FileOffset,
         /// exclusive
         file_offset_end: FileOffset,
-    ) !CachedRead {
+    ) ReadIoUringError!ReadHandle {
         if (!use_io_uring) @compileError("io_uring disabled");
+        const threadlocal_io_uring = try io_uring();
 
-        const frame_indices = try self.computeFrameIndices(
-            file_id,
+        const frame_refs = try self.manager.getAllIndices(
             allocator,
+            file_id,
             file_offset_start,
             file_offset_end,
         );
-        errdefer allocator.free(frame_indices);
+        errdefer allocator.free(frame_refs);
+        errdefer self.manager.getAllIndicesRollback(frame_refs);
+
+        var i: u32 = 0;
+        var n_read: u32 = 0;
+        while (i < frame_refs.len or n_read < frame_refs.len) {
+            var queue_full = false;
+
+            if (i < frame_refs.len) {
+                const frame_ref = &frame_refs[i];
+
+                const populated = self.manager.contains_valid_data[frame_ref.f_idx].load(.acquire);
```

Comment:
probs need to change/double check the var names match the field name changes

Reply:
Removed / gave it a double check https://github.com/Syndica/sig/commit/f40689214b7f8a0945d24c5a2847c29af6db0203